src/seaport.ts

```diff
@@ -932,4 +943,27 @@ export class Seaport {
       domain
     );
   }
+
+  public setDomain(
+    domain: string,
+    accountAddress: string
+  ): TransactionMethods<
+    ContractMethodReturnType<DomainRegistryContract, "setDomain">
+  > {
+    const signer = this._getSigner(accountAddress);
+
+    return getTransactionMethods(
+      this.domainRegistry.connect(signer),
+      "setDomain",
+      [domain]
+    );
+  }
+
+  public getDomain(tag: string, index: number): Promise<string> {
+    return this.domainRegistry.getDomain(tag, index);
+  }
+
+  public async getDomains(tag: string): Promise<string[]> {
+    return this.domainRegistry.getDomains(tag);
```

Comment:
For this one, I'd say try/catch it and if it fails then fall back to getTotalDomains + map(getDomain)